File: benchmarks/poly/logparse/logparse.py
from collections import defaultdict
import sys, time
_t0 = time.perf_counter_ns()
statuses = ["200", "200", "200", "404", "500"]
cnt = defaultdict(int)
lat = defaultdict(int)
for i in range(150000):
    path = "/api/" + str(i % 50)
    status = statuses[i % 5]
    line = "GET " + path + " " + status + " " + str(i % 250)
    f = line.split(" ")
    cnt[f[2]] += 1
    lat[f[1]] += int(f[3])
for k in sorted(cnt.keys()):
    print(k + " " + str(cnt[k]))
print(sum(lat.values()))
print(f"bench_ns={time.perf_counter_ns() - _t0}", file=sys.stderr)

File: benchmarks/poly/logparse/logparse.rs
use std::collections::HashMap;

fn main() {
    let t0 = std::time::Instant::now();
    let statuses = ["200", "200", "200", "404", "500"];
    let mut cnt: HashMap<String, i64> = HashMap::new();
    let mut lat: HashMap<String, i64> = HashMap::new();
    for i in 0..150_000 {
        let path = format!("/api/{}", i % 50);
        let status = statuses[i % 5];
        let line = format!("GET {} {} {}", path, status, i % 250);
        let f: Vec<&str> = line.split(' ').collect();
        *cnt.entry(f[2].to_string()).or_insert(0) += 1;
        let n: i64 = f[3].parse().unwrap_or(0);
        *lat.entry(f[1].to_string()).or_insert(0) += n;
    }
    let mut keys: Vec<&String> = cnt.keys().collect();
    keys.sort();
    for k in keys {
        println!("{} {}", k, cnt[k]);
    }
    let total: i64 = lat.values().sum();
    println!("{}", total);
    eprintln!("bench_ns={}", t0.elapsed().as_nanos());
}
